src/resolvers/donationResolver.ts

```diff
@@ -175,12 +176,13 @@ export class DonationResolver {
     const fromWalletAddressesArray: string[] = fromWalletAddresses.map(o =>
       o.toLowerCase(),
     );
-    const donations = await this.donationRepository.find({
-      where: {
-        fromWalletAddress: In(fromWalletAddressesArray),
-      },
-    });
-    return donations;
+    return this.donationRepository
+      .createQueryBuilder('donation')
+      .where({ fromWalletAddress: In(fromWalletAddressesArray) })
+      .leftJoin('donation.user', 'user')
+      .addSelect(publicSelectionFields)
+      .leftJoinAndSelect('donation.project', 'project')
```

Comment:
I am worried project admin sensitive data be loaded here because of that `{eagher: true}` option in src/entities/project.ts

Reply:
Hey Amin. I guess the ``` { eager: true } ``` is being used on our tests for easy loading. They are loaded when using repository like ``` .find() ``` methods.

Since we use querybuilder on the mutations it won't load the extra data.

https://orkhan.gitbook.io/typeorm/docs/eager-and-lazy-relations

```
Eager relations only work when you use find* methods. If you use QueryBuilder eager relations are disabled and have to use leftJoinAndSelect to load the relation. Eager relations can only be used on one side of the relationship, using eager: true on both sides of relationship is disallowed.
```